src/main.rs

```diff
@@ -166,18 +166,43 @@ fn main() {
         libc::signal(SIGINT, mem::transmute(handle_sigint));
     }
 
-    let format = |record: &LogRecord| {
-        let t = time::now();
-        format!("{}.{:03}: {}{:5} {}",
-            time::strftime("%Y-%m-%d %H:%M:%S", &t).unwrap(),
-            t.tm_nsec / 1000_000,
-            tid_str(),
-            record.level(),
-            record.args()
-        )
-    };
     let mut builder = LogBuilder::new();
-    builder.format(format).filter(None, LogLevelFilter::Info);
+    let istty = unsafe { libc::isatty(libc::STDERR_FILENO as i32) } != 0;
+    if istty {
+        // Colorized output formatter
+        let format = |record: &LogRecord| {
+            let t = time::now();
+            let level_color = match record.level() {
+                log::LogLevel::Error => "\x1b[1;31m",  // bold red
+                log::LogLevel::Warn  => "\x1b[1;33m",  // bold yellow
+                log::LogLevel::Info  => "\x1b[1;32m",  // bold green
+                log::LogLevel::Debug => "\x1b[1;34m",  // bold blue
+                log::LogLevel::Trace => "\x1b[1;35m"   // bold magenta
```

Comment:
Ahh - ok - I missed that.

I'm used to the colored-logcat output for adb, which assigns a color to each PID and colorizes the left portion of the display using that, and then colors the right portion using the log level.
https://www.dropbox.com/s/2bxgmp9wzh0r0um/Screenshot%20from%202016-03-15%2019%3A11%3A17.png?dl=0
The colored logcat script is here: https://github.com/dhylands/colored-logcat